.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+name: release
+
+on:
+  pull_request:
+    branches:
+    - main
+  push:
+    branches:
+    - main
+    tags:
+    - "v*.*.*"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        platform:
+        - os: ubuntu-20.04
+          target: x86_64-unknown-linux-musl
+        - os: macos-11
+          target: x86_64-apple-darwin
+        # TODO: provide aarch64 builds
```

Comment:
How about macOS [fat binary](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) with the `lipo` tool?

Reply:
Is it reproducible?